packages/router/utils.ts

```diff
@@ -1209,37 +1209,16 @@ export function resolveTo(
   // to the current location's pathname and *not* the route pathname.
   if (toPathname == null) {
     from = locationPathname;
-  } else if (isPathRelative) {
-    let fromSegments =
-      routePathnames.length === 0
-        ? []
-        : routePathnames[routePathnames.length - 1]
-            .replace(/^\//, "")
-            .split("/");
-
-    if (toPathname.startsWith("..")) {
-      let toSegments = toPathname.split("/");
-
-      // With relative="path", each leading .. segment means "go up one URL segment"
-      while (toSegments[0] === "..") {
-        toSegments.shift();
-        fromSegments.pop();
-      }
-
-      to.pathname = toSegments.join("/");
-    }
-
-    from = "/" + fromSegments.join("/");
   } else {
     let routePathnameIndex = routePathnames.length - 1;
 
-    if (toPathname.startsWith("..")) {
+    // With relative="route" (the default), each leading .. segment means
+    // "go up one route" instead of "go up one URL segment".  This is a key
+    // difference from how <a href> works and a major reason we call this a
+    // "to" value instead of a "href".
+    if (!isPathRelative && toPathname.startsWith("..")) {
```

Comment:
We just need to skip processing any `..` segments in `to` when generating our `from` pathname when `relative="path"`